packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -20,6 +21,37 @@ import 'ios/plist_utils.dart' as plist;
 import 'project.dart';
 import 'tester/flutter_tester.dart';
 
+class ApplicationPackageFactory {
+  static ApplicationPackageFactory get instance => context[ApplicationPackageFactory];
+
+  Future<ApplicationPackage> getApplicationPackageForPlatform(
```

Comment:
This method name seems a little overly verbose now that it's in a `ApplicationPackageFactory` class.  Perhaps just `getPackageForPlatform()`?